controllers/front/ProductController.php

```diff
@@ -151,6 +151,9 @@ public function init()
                 header('HTTP/1.1 403 Forbidden');
                 header('Status: 403 Forbidden');
                 $this->errors[] = $this->trans('You do not have access to this product.', array(), 'Shop.Notifications.Error');
+                $this->setTemplate('errors/forbidden');
+
+                return;
```

Comment:
Yes, but are you sure that there will be never else after that ? Is the return really a big thing/problem to leave it, to be sure about it's need ? ;-)